src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/ServiceApiImpl.java

```diff
@@ -32,10 +37,24 @@
         if (isNotEmpty(service.getServiceCharacteristic())) {
             service.getServiceCharacteristic().forEach(ch -> params.put(ch.getName(), ch.getValue()));
         }
+
+        //if service processes LEP`s asynchronously, RequestContextHolder will not be able to get request
+        //attributes from thread that will execute LEP
+        ServletRequestAttributes requestAttributes = (ServletRequestAttributes) RequestContextHolder.getRequestAttributes();
```

Comment:
maybe better to extract function like `addAllHeaders` ?

Reply:
agree, done